Avoid overlinking abseil in libs using re2

re2 exports a pkg-config file that instructs all users of re2 to link
against some libraries from libabseil when linking against re2. Because
abseil often changes its soversion, this requires many ports that use
re2 to be revbumped whenever abseil changes, even though they only use
symbols from re2.

Avoid this by using the Requires.private field in the re2 pkg-config
file as documented in [1]. See also [2], which explains the difference
in linking.

[1]: https://people.freedesktop.org/~dbn/pkg-config-guide.html#concepts
[2]: https://stackoverflow.com/a/61674208

Upstream-Status: Pending
--- ./re2.pc.in.orig	2023-06-30 16:48:20.000000000 +0200
+++ ./re2.pc.in	2023-08-26 21:36:05.000000000 +0200
@@ -3,7 +3,7 @@
 
 Name: re2
 Description: RE2 is a fast, safe, thread-friendly regular expression engine.
-Requires: @REQUIRES@
+Requires.private: @REQUIRES@
 Version: @SONAME@.0.0
 Cflags: -pthread -I${includedir}
 Libs: -pthread -L${libdir} -lre2
